### PR TITLE
Document two new feature gates around CEL cost enforcement

### PR DESCRIPTION
**Pull request by cici37**
### Description
related PR: https://github.com/kubernetes/kubernetes/pull/124675

### Issue

Closes: #

### CODE REVIEW

**Comment by cici37** [comment]
/assign @kubernetes/sig-docs-en-owners
for approval. Thanks

**Comment by kundan2707** [comment]
[Preview](https://deploy-preview-47321--kubernetes-io-main-staging.netlify.app/docs/reference/command-line-tools-reference/feature-gates/)

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  list: never
+  render: false
+
+stages:
+  - stage: alpha
+    defaultValue: false
+    fromVersion: "1.31"
+    
+---
+Enable the enforcement of the strict cost calculation for CEL extended libraries in ValidatingAdmissionPolicy.
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md:14]: Even better:
```suggestion
Enable enforcement of the strict cost calculation for CEL extended libraries used within a
[ValidatingAdmissionPolicy](/docs/reference/access-authn-authz/validating-admission-policy/).
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-webhooks.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  list: never
+  render: false
+
+stages:
+  - stage: alpha
+    defaultValue: false
+    fromVersion: "1.31"
+    
+---
+Enable the enforcement of the strict cost calculation for CEL extended libraries in `matchConditions` field under Webhooks.
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-webhooks.md:14]: Even better
```suggestion
Enable enforcement of the strict cost calculation for CEL extended libraries within the
`matchConditions` field for
[admission webhooks](/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks).
```

**Review by sftim — COMMENTED** [review]
LGTM for docs. Thanks for proposing docs before the release.

Optional tweaks available.

**Comment by sftim** [comment]
/retitle Document two new feature gates around CEL cost enforcement

**Comment by sftim** [comment]
/sig api-machinery

/lgtm

These are good to ship as they are; docs meet our quality bar for alpha.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md (hunk 1 above)
Comment (cici37) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md:14]: Thank you. Updated :)

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-webhooks.md (hunk 2 above)
Comment (cici37) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-webhooks.md:14]: Thanks! Updated :)

**Comment by cici37** [comment]
Addition info: we backported those feature gates into 1.30.x patch releases so I updated the fromVersion to 1.30. Thanks

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+title: StrictCostEnforcementForVAP
+content_type: feature_gate
+_build:
+  list: never
+  render: false
+
+stages:
+  - stage: alpha
```
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md:9]: Is this an _alpha_ or _beta_ feature? Looking at the code, it appears to be beta. Apologies if I misunderstood or if there's any confusion on my end.

	genericfeatures.StrictCostEnforcementForVAP: {Default: false, PreRelease: featuregate.Beta},
	
_[Referring to kubernetes/kubernetes/pkg/features/kube_features.go#L1300 (here)](https://github.com/kubernetes/kubernetes/blob/6fc0a69044f1ac4c13841ec4391224a2df241460/pkg/features/kube_features.go#L1300)_

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    defaultValue: false
+    fromVersion: "1.30"
```
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md:11]: If these features were introduced in v1.30, we might want to target the _main_ branch for this PR, which currently has the latest v1.30 docs?

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-webhooks.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+---
+title: StrictCostEnforcementForWebhooks
+content_type: feature_gate
+_build:
+  list: never
+  render: false
+
+stages:
+  - stage: alpha
```
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-webhooks.md:9]: Similarly, whether this `StrictCostEnforcementForWebhooks` feature gate is in the alpha or beta stage?

	genericfeatures.StrictCostEnforcementForWebhooks: {Default: false, PreRelease: featuregate.Beta},
	
[Referring to kubernetes/kubernetes/pkg/features/kube_features.go#L1302 (here)](https://github.com/kubernetes/kubernetes/blob/6fc0a69044f1ac4c13841ec4391224a2df241460/pkg/features/kube_features.go#L1302)

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md (hunk 4 above)
Comment (cici37) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md:11]: Makes sense. I'll close this one and create another PR against the main branch then. Thanks!

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md (hunk 3 above)
Comment (cici37) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md:9]: Ohh I forgot we agreed to introduce this as beta for this special fg. Thanks

**Comment by cici37** [comment]
/close 
in favor of https://github.com/kubernetes/website/pull/47326

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md (hunk 4 above)
Comment (sftim) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-VAP.md:11]:
```suggestion
    fromVersion: "1.31"
```
?

File: content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-webhooks.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+title: StrictCostEnforcementForWebhooks
+content_type: feature_gate
+_build:
+  list: never
+  render: false
+
+stages:
+  - stage: beta
+    defaultValue: false
+    fromVersion: "1.30"
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/strict-cos-enforcement-for-webhooks.md:11]:
```suggestion
    fromVersion: "1.31"
```
?

**Pull request closed without merging**